sensors/ObstacleSensor.py

```diff
@@ -0,0 +1,53 @@
+from sensors.SensorCommon import Sensor
+from sensors.SensorCommon import Dist2Circ
+import numpy as np
+import operator
+
+
+class ObstacleSensor(Sensor):
+
+    def __init__(self, obstacles=False, nbObstacles=0, SensorRange=10, mode="position"):
+        super().__init__(nbObservations=nbObstacles, SensorRange=SensorRange)
+        ## todo: implement obstacle and goal flag.
+        self._Observation = np.ones([self._nbObservations, 2]) * self._limSensor
+        self._mode = mode
+        self._setSensorName()
+
+    def _setSensorName(self):
+        if self._mode == "position":
+            self._name = "ObstaclePseudoSensorPosition"
+        elif self._mode == "distance":
+            self._name = "ObstaclePseudoSensorDistance"
```

Comment:
This name is a bit too long for a dict-key. Also for consistency with respect to the GoalSensor you can call it `ObstaclesPositions`.